src/IO/Process/InteractiveProcess.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Dock\IO\Process;
+
+use Dock\IO\Process\Pipe\Pipe;
+use Dock\IO\Process\WaitStrategy\WaitStrategy;
+use Symfony\Component\Process\Process;
+
+class InteractiveProcess
+{
+    /**
+     * @var Pipe
+     */
+    private $pipe;
+    /**
+     * @var WaitStrategy
+     */
+    private $waitStrategy;
+    /**
+     * @var Process
+     */
+    private $process;
+
+    public function __construct(Process $process, Pipe $pipe, WaitStrategy $waitStrategy)
+    {
+        $this->pipe = $pipe;
+        $this->waitStrategy = $waitStrategy;
+        $this->process = $process;
+    }
+
+    public function updatePipe(Pipe $pipe)
+    {
+        $this->pipe = $pipe;
+    }
+
+    public function updateWaitStrategy(WaitStrategy $waitStrategy)
```

Comment:
Can't be done at runtime so shouldn't be here. Using more the builder to setup these strategy would be better.